### ISSUE TITLE
AssetResource readText() doesn't work on JVM

### ISSUE THREAD

**Issue by wakaztahir** [icerockdev/moko-resources#424]
this code is not working in compose for desktop JVM , while the code works on Android , cannot escape backslashes for assets in JS

```kotlin
@Composable
actual fun loadAsText(asset: AssetResource): String {
    return produceState("") {
        value = asset.readText()
    }.value
}
```

It throws the exception

![Capture](https://user-images.githubusercontent.com/42442700/221343191-681fdc68-1fc9-4023-9d4d-e774aa463ac8.PNG)

```
Exception in thread "AWT-EventQueue-0" java.io.FileNotFoundException: Couldn't open resource as stream at: files\export-from-sketchable.md
	at dev.icerock.moko.resources.FileResource.readText(FileResource.kt:15)
	at com.wakaztahir.resourceshelper.DesktopResourcesKt$loadAsText$1.invokeSuspend(DesktopResources.kt:15)
	at com.wakaztahir.resourceshelper.DesktopResourcesKt$loadAsText$1.invoke(DesktopResources.kt)
	at com.wakaztahir.resourceshelper.DesktopResourcesKt$loadAsText$1.invoke(DesktopResources.kt)
	at androidx.compose.runtime.SnapshotStateKt__ProduceStateKt$produceState$1.invokeSuspend(ProduceState.kt:86)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher$dispatch$2$1.invoke(FlushCoroutineDispatcher.skiko.kt:62)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher$dispatch$2$1.invoke(FlushCoroutineDispatcher.skiko.kt:57)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher.performRun(FlushCoroutineDispatcher.skiko.kt:91)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher.access$performRun(FlushCoroutineDispatcher.skiko.kt:37)
	at androidx.compose.ui.platform.FlushCoroutineDispatcher$dispatch$2.invokeSuspend(FlushCoroutineDispatcher.skiko.kt:57)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1080)
	at java.desktop/java.awt.Component.show(Component.java:1729)
	at java.desktop/java.awt.Component.setVisible(Component.java:1676)
	at java.desktop/java.awt.Window.setVisible(Window.java:1036)
	at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1016)
	at androidx.compose.ui.window.AwtWindow_desktopKt$AwtWindow$4$1.invokeSuspend(AwtWindow.desktop.kt:120)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [androidx.compose.ui.awt.ComposeLayer$coroutineExceptionHandler$1@2636630f, androidx.compose.runtime.BroadcastFrameClock@567e5c16, StandaloneCoroutine{Cancelling}@76e10a52, FlushCoroutineDispatcher@52480367]

```

**Comment by wakaztahir** [comment]
The files have been copied , I don't know why It can't read the text

![Capture](https://user-images.githubusercontent.com/42442700/221343365-d8d353a2-4cb6-4489-bec1-2949a9ccd734.PNG)

**Comment by wakaztahir** [comment]
I have found the cause on windows , see on windows the file separator character returns '\' which for some reason causes this problem

the problem is with the following code

```
    return AssetResource(
        resourcesClassLoader,
        originalPath,
        "files${File.separatorChar}$originalPath"
    )
```
The workaround is
    
 ```kotlin
val stream = asset.resourcesClassLoader.getResourceAsStream(asset.filePath.replace('\\', '/'))
            ?: throw FileNotFoundException("Couldn't open resource as stream at: ${asset.filePath}")
        value = stream.use { it.readBytes().decodeToString() }
```

boom your assets are working

**Comment by wakaztahir** [comment]
since forward slash is working , referencing this stackoverflow post , I'd suggest to change File.separator to hardcoded '/' since it will work on both platforms , linux and windows
https://stackoverflow.com/questions/2417485/difference-between-file-separator-and-slash-in-paths

**Comment by wakaztahir** [comment]
To solidify the usage of hardcoded "/"
Java Requires Forward Slash instead of Backslash on Windows In Resources
https://stackoverflow.com/questions/72951119/java-requires-slash-instead-of-backslash-on-windows-in-resources

**Comment by wakaztahir** [comment]
https://github.com/icerockdev/moko-resources/pull/428 changes have been made , ready to merge

**Comment by Alex009** [comment]
will be released in 0.21.0
